extras/foreman_protector/dnf/foreman-protector.py

```diff
@@ -0,0 +1,76 @@
+import dnf
+import dnf.exceptions
+from dnfpluginscore import _, logger
+
+import configparser
+package_whitelist = set()
+
+class ForemanProtector(dnf.Plugin):
+    name = 'foreman-protector'
+    config_name = 'foreman-protector'
+
+    def __init__(self,base,cli):
+        self.base = base
+        self.cli = cli
+
+    def config(self):
+        global fileurl
```

Comment:
I don't like globals, but I guess this is some kind of `dnf` convention when it comes to passing data between things?

Reply:
There are some use cases for `global` in dnf plugins. There are two "flavours" of dnf plugins:

1. plugins that change behaviour of dnf globally in some sense (for example this foreman-protector)
2. plugins that provide new command

And sometimes (like in dnf-plugins-core/plugins/versionlock.py) you need to share some data between both types (they are represented by separate classes). Then the global may be useful.

But here I do not see the reason for global, fileurl could be member of ForemanProtector class.